src/Microsoft.AspNet.Razor/RazorError.cs

```diff
@@ -77,7 +77,8 @@ public bool Equals(RazorError other)
         {
             return other != null &&
                 string.Equals(other.Message, Message, StringComparison.Ordinal) &&
-                Location.Equals(other.Location);
+                Location.Equals(other.Location) &&
+                Length.Equals(other.Length);
```

Comment:
if you need this for test purposes, use a test-specific `IEqualityComparer<RazorError>` instead